src/lib/select/select.ts

```diff
@@ -242,6 +245,15 @@ export class MdSelect implements AfterContentInit, ControlValueAccessor, OnDestr
   ngAfterContentInit() {
     this._initKeyManager();
     this._resetOptions();
+
+    // Assign any values that were deferred until the component is initialized.
+    if (this._tempValue) {
+      Promise.resolve(null).then(() => {
+        this.writeValue(this._tempValue);
+        this._tempValue = null;
+      });
+    }
+
     this._changeSubscription = this.options.changes.subscribe(() => {
```

Comment:
Given that we are doing the same thing on option changes (i.e. writing the value next tick), I think it might make more sense to just incorporate the logic into the existing stream with `startWith()`. `this._control.value` should have the correct value already, so saving a temp version shouldn't be necessary.